Khiem_Nguyen/Python_3/Python/python_stack/python_fundamentals/for_loop_basic2.py

```diff
@@ -0,0 +1,103 @@
+#1
+def makeItBig(arr):
+    for i in range(0,len(arr)):
+        if (arr[i]>0):
+            arr[i]="big"
+    print(arr)
+    return(arr)
+makeItBig([-1,2,5,-5])
+
+#2
+def countPositives(arr):
+    count = 0
+    for i in range(0,len(arr)):
+        if arr[i]>0:
+            count = count + 1
+    arr[len(arr)-1] = count
+    print(arr)
+countPositives([-1,1,1,1])
+
+#3
+def sumTotal(arr):
+    sum = 0
+    for i in range(0,len(arr)):
+        sum = sum + arr[i]
+    print(sum)
+sumTotal([1,2,3,4])
+
+#4
+def average(arr):
+    sum = 0
+    for i in range(0,len(arr)):
+        sum = sum + arr[i]
+    average = sum/len(arr)
+    print(sum)
+    print(len(arr))
+    print(average)
+average([1,2,3,4])
+
+#5
+def length(arr):
+    print(len(arr))
+length([1,2,3,4])
+
+#6
+def minimum(arr):
+    if len(arr) == 0:
+        return False
+    currentMin = arr[0]
+    for i in range(0,len(arr)):
+        if arr[i] < currentMin:
+            currentMin = arr[i]
+    print(currentMin);
+minimum([1,2,3,4])
+minimum([-1,-2,-3])
+
+#7
+def maximum(arr):
+    if len(arr) == 0:
+        return False
+    currentMax = arr[0]
+    for i in range(0,len(arr)):
+        if arr[i] > currentMax:
+            currentMax = arr[i]
+    print(currentMax)
+print(maximum([]))
+maximum([1,2,3,4])
+maximum([-1,-2,-3])
+
+#8
+def analyze(arr):
+    myDict = {"sumTotal":"", "average":"", "minimum":"", "maximum":"", "length":""}
+    sum = 0
+    for i in range(0,len(arr)):
+        sum = sum + arr[i]
+        myDict["sumTotal"] = sum
+        average = sum/len(arr)
+        myDict["average"] = average
+        currentMin = arr[0]
+        for i in range(0,len(arr)):
```

Comment:
You need not use nested loops. It's expensive. Draw out a T diagram for this code and refactor it!

Reply:
I removed the two "for" loops nested. I ran it and it's good now. I will commit again with changes. Thanks for feedback!